File: src/main/java/nl/shadowlink/tools/shadowlib/ipl/Item_BLOK.java
package nl.shadowlink.tools.shadowlib.ipl;

import nl.shadowlink.tools.io.ReadFunctions;
import nl.shadowlink.tools.shadowlib.utils.GameType;
import nl.shadowlink.tools.shadowmapper.utils.hashing.HashTable;

import java.util.logging.Level;
import java.util.logging.Logger;

/**
 * @author Shadow-Link
 */
public class Item_BLOK extends IPL_Item {
    private GameType gameType;

    Item_BLOK(GameType gameType) {
        this.gameType = gameType;
    }

    @Override
    public void read(String line) {
        //Message.displayMsgHigh(line);
    }

    @Override
    public void read(ReadFunctions rf) {
        throw new UnsupportedOperationException("Not supported yet.");
    }

    @Override
    public void read(ReadFunctions rf, HashTable hashTable) {
        Logger.getLogger("IPL").log(Level.INFO, getClass().getSimpleName() + " not supported yet.");
    }

}

File: src/main/java/nl/shadowlink/tools/shadowlib/ipl/Item_BLOK.kt
package nl.shadowlink.tools.shadowlib.ipl

import nl.shadowlink.tools.io.ReadFunctions
import nl.shadowlink.tools.shadowlib.utils.GameType
import nl.shadowlink.tools.shadowlib.utils.hashing.HashTable
import java.util.logging.Level
import java.util.logging.Logger

/**
 * @author Shadow-Link
 */
class Item_BLOK internal constructor(private val gameType: GameType) : IplItem() {
    override fun read(line: String) {
        //Message.displayMsgHigh(line);
    }

    override fun read(rf: ReadFunctions) {
        throw UnsupportedOperationException("Not supported yet.")
    }

    override fun read(rf: ReadFunctions, hashTable: HashTable) {
        Logger.getLogger("IPL").log(Level.INFO, javaClass.simpleName + " not supported yet.")
    }
}